src/CarbonAware.CLI/test/integrationTests/IntegrationTestingBase.cs

```diff
@@ -0,0 +1,130 @@
+﻿using CarbonAware.DataSources.Configuration;
+using CarbonAware.DataSources.Mocks;
+using CarbonAware.DataSources.Json.Mocks;
+using CarbonAware.DataSources.WattTime.Mocks;
+using NUnit.Framework;
+using System.CommandLine.IO;
+using System.Diagnostics;
+
+namespace CarbonAware.CLI.IntegrationTests;
+
+/// <summary>
+/// A base class that does all the common setup for the Integration Testing
+/// Overrides WebAPI factory by switching out different configurations via _datasource
+/// </summary>
+public abstract class IntegrationTestingBase
+{
+    internal DataSourceType _dataSource;
+    internal string? _dataSourceEnv;
+    protected IDataSourceMocker _dataSourceMocker;
+    protected TestConsole _console = new();
+
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    public IntegrationTestingBase(DataSourceType dataSource)
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    {
+        _dataSource = dataSource;
+        _dataSourceEnv = Environment.GetEnvironmentVariable("CarbonAwareVars__CarbonIntensityDataSource");
+    }
+
+    protected int InvokeCLI(string arguments)
+    {
+        // Initialize process here
+        Process proc = new Process();
+        proc.StartInfo.FileName = "caw.exe";
+        // add arguments as whole string
+        proc.StartInfo.Arguments = arguments;
+
+        // use it to start from testing environment
+        proc.StartInfo.UseShellExecute = false;
+
+        // redirect outputs to have it in testing console
+        proc.StartInfo.RedirectStandardOutput = true;
+        proc.StartInfo.RedirectStandardError = true;
+
+        // set working directory
+        proc.StartInfo.WorkingDirectory = Environment.CurrentDirectory;
+
+        Console.SetOut(_console.Out.CreateTextWriter());
+        Console.SetError(_console.Error.CreateTextWriter());
+
+        // start and wait for exit
+        proc.Start();
+        proc.WaitForExit();
+
+        // get output to testing console.
+        Console.WriteLine(proc.StandardOutput.ReadToEnd());
+        Console.Write(proc.StandardError.ReadToEnd());
+
+        // reset Console Streams
+        var standardOutput = new StreamWriter(Console.OpenStandardOutput());
+        var standardError = new StreamWriter(Console.OpenStandardError());
+        standardOutput.AutoFlush = true;
+        standardError.AutoFlush = true;
+        Console.SetOut(standardOutput);
+        Console.SetError(standardError);
+
+        // return exit code
+        return proc.ExitCode;
```

Comment:
nit: you can do an assert here like:
```c#
Assert.AreEqual(0, process.ExitCode, $"Process exited unsuccessfully with Exit Code {process.ExitCode}");
```
and you can save all the lines that checks for error != 0 on all the other tests.

Reply:
I want to leave that up to the tests themselves.  We likely want to test other exit codes depending on the test scenario.